### PR TITLE
feat: readiness probe

### PR DESCRIPTION
**Pull request by jrriehl**
Implements readiness probe proposed here: https://github.com/fetchai/uAgents/discussions/183.

Here's what this looks like from a `curl` command:
```bash
$ curl -I -H "x-uagents-address: agent1q2kxet3vh0scsf0sm7y2erzz33cve6tv5uk63x64upw5g68kr0chkv7hw50" http://127.0.0.1:8001/submit
HTTP/1.1 200 OK
date: Tue, 24 Oct 2023 13:17:35 GMT
server: uvicorn
x-uagents-status: ready
x-uagents-response-time-hint: 5
Transfer-Encoding: chunked
```

### CODE REVIEW

**Review by Archento — CHANGES_REQUESTED** [review]
The response setup for the agentverse looks good and the tests are appropriate. I did find an occasion during manual testing though and I am unsure on how detailed the if-else branches should be (we can't catch every case)
So once the two questions are resolved it's green from my side!

File: python/src/uagents/asgi.py [hunk 1]
```diff
@@ -110,40 +194,13 @@ async def __call__(
...
-                        ],
-                    }
-                )
-                await send(
-                    {
-                        "type": "http.response.body",
-                        "body": b'{"error": "missing header: content-type"}',
-                    }
-                )
+            await self.handle_missing_content_type(headers, send)
```
[the remark this answers is not in the document]
Comment (jrriehl) [reply python/src/uagents/asgi.py:199]: Fixed, good catch!

File: python/src/uagents/config.py [hunk 2]
```diff
@@ -37,6 +37,7 @@
 ALMANAC_API_URL = AGENTVERSE_URL + "/v1/almanac/"
 MAILBOX_POLL_INTERVAL_SECONDS = 1.0
 
+RESPONSE_TIME_HINT_SECONDS = 5
```
[the remark this answers is not in the document]
Comment (ejfitzgerald) [reply python/src/uagents/config.py:40]: I think this is a reasonable default for the hint response time. I guess from my side the intention here is that this hint value is just a indication (nothing more) about the liveness of the agent. In more IoT scenarios this hint could be much longer.  I guess we should use this a rough hint as to what timeout value we should have for waiting for a response.

**Review by Archento — APPROVED** [review]
I'm good with the current state 👍🏻

**Pull request merged**